### PR TITLE
Feature/directions form

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Wires up the directions form controls on the home page:
  - Adds back address autocomplete (needs styling; see #571)
  - Wires up new geolocation button
  - Updates handing of mode selection and toggling classes on the new mode buttons
  - Transitions to the map view when both 'from' and 'to' fields completed (does not query for directions)

### CODE REVIEW

File: src/app/scripts/cac/control/cac-control-bike-mode-options.js [hunk 1]
```diff
@@ -67,12 +69,26 @@ CAC.Control.BikeModeOptions = (function ($) {
...
      *
-     * * @param modeSelectors {String} jQuery selector like '#someId input'
+     * @param modeSelectors {String} jQuery selector like '#someId input'
+     * @returns {String} comma-separated list of OpenTripPlanner mode parameters
      */
     function getMode(modeSelectors) {
-        var mode = $(modeSelectors + options.modes.walkBike + ':checked').val() || 'WALK';
-        var transit = $(modeSelectors + options.modes.transit).prop('checked');
-        mode = transit ? 'TRANSIT,' + mode : mode;
+        var mode = '';
```
Comment (KlaasH) [inline src/app/scripts/cac/control/cac-control-bike-mode-options.js]: A more heavily lodashified version of this function:
```
var $selected = $(modeSelectors);
if (!$selected) {
    console.error('no mode controls found');
    return options.defaultMode;
}
var mode = _(options.modes).filter(function(val, key) { return $selected.hasClass(key); }).join(',');
return mode || options.defaultMode;
```

**Comment by flibbertigibbet** [comment]
updated

File: src/app/scripts/cac/pages/cac-pages-home.js [hunk 2]
```diff
@@ -158,7 +205,7 @@ CAC.Pages.Home = (function ($, BikeModeOptions,  MapControl, Templates, UserPref
         event.preventDefault();
 
         // hide existing destinations list and show loading spinner
-        $(options.selectors.destinationsContainer).addClass('hidden');
+        $(options.selectors.placeList).addClass('hidden');
```
Comment (KlaasH) [inline src/app/scripts/cac/pages/cac-pages-home.js]: There's no "View all destinations" link on the redesigned home page.  Is there a reason to keep this function (and the related selector and listener) around?

Comment (flibbertigibbet) [inline src/app/scripts/cac/pages/cac-pages-home.js]: There is a bunch of old code that has been retained. I've been holding off on purging things generally until some more of the new functionality is stable, as some of it we do want or will break something else if removed immediately.

**Pull request merged**